# Hi, I'm Lawrence Fung

I work in bioinformatics and computational biology, with a focus on data analysis, statistical modeling, and reproducible workflows applied to genomics and human healthâ€“related datasets. My projects emphasize transforming high-dimensional biological data into interpretable insights using R and Python.

## Featured Projects

- **[NGS Normalization](https://github.com/funghub/gene_expression_data-normalization)**  
  This R notebook normalizes the raw counts with TMM (trimmed mean of M-values) and CPM (counts per million) from Pasilla gene expression. A heatmap via a distance matrix is generated with pheatmap library. With edgeR, plots of normalized values cpm & tmm are created. With dplyr, tidyr, and ggplot2 libraries a density plot of expression values for all samples is created.
  
- **[NGS Clustering](https://github.com/funghub/gene_expression_data-clustering)**  
  This R notebook will use statistical techniques to understand a NGS data set and compare gene expression by hierarchical clustering and k-means clustering. Explanations of the diagrams are included in the notebook.
  
  This code will use statistical techniques to understand a NGS data set and compare gene expression by hierarchical clustering and k-means clustering. 

- **[NGS Classification](https://github.com/funghub/gene_expression_data-classification)**  
  This R notebook takes a NGS data set and applies a binary classification method to sort out replicate1 and other replicates. Then this is plot on a ROC curve to explore different thresholds. The notebook also explores the use of confusion matrix to calculate specificity (tnr), sensitivity (tpr), and false positve rate. Explanations of the diagrams are included in the notebook.

- **[Hidden Markov Model Protein Family](https://github.com/funghub/proteinfamHMM)**  
  This R notebook creates a HMM model that represents a given diagram showing all the path of the hidden states. I will find the most likely sequence of hidden states for each protein family. In addition, given a sequence of observations, I will also find the likely family that would have generated those observations and then find the most likely sequence of hidden states of that family with the Viterbi algorithm.

- **[Converting HGVS Notation to Protein Sequences](https://github.com/funghub/cysticfibrosis-project)**  
  Through CFTR-Fr datasets, we seek to visualize the list of mutations on AlphaFold by inputting the DNA sequences and protein sequences created from the HGVS nomenclature and reference gene on NCBI with Python. We investigate how mutations in those regions of the gene affect the final shape and function of the CFTR protein, specifically the crucial parts of the protein, such as Phe508del in 90% of cystic fibrosis patients.
  
## Connect With Me

- LinkedIn: (https://www.linkedin.com/in/funglawrence/)
